api/app/src/main/kotlin/packit/service/PacketService.kt

```diff
@@ -121,12 +121,18 @@ class BasePacketService(
     }
 
     override fun streamZip(paths: List<String>, id: String, output: OutputStream) {
-        val hashesByPath = getMetadataBy(id).files
+        val metadataFiles = getMetadataBy(id).files
+        val hashesByPath = metadataFiles
             .filter { it.path in paths }
             .associateBy({ it.path }, { it.hash })
+        val notFoundPaths = paths.filter { path -> metadataFiles.none { it.path == path } }
 
-        if (hashesByPath.size != paths.size) {
-            throw PackitException("notAllFilesFound", HttpStatus.NOT_FOUND)
+        if (notFoundPaths.isNotEmpty()) {
+            throw PackitException("notAllFilesFound: ${notFoundPaths.joinToString(", ")}", HttpStatus.NOT_FOUND)
+        }
+
+        if (paths.isEmpty()) {
+            throw PackitException("noFilesProvided", HttpStatus.BAD_REQUEST)
```

Comment:
Sorry, I should have spotted before that "notAllFilesFound", and now "noFilesProvided" need to be added to the resource bundle (`src/main/resources/errorBundle.properties`) so they can be mapped to friendlier error messages (which in theory may eventually include non-English) translations.  However, we don't currently support error messages in the bundle which can be interpolated as you're doing here with the file names, but that would be fairly easy to add in - would suggest doing that in another ticket and just using a friendly error messages here for `notAllFilesFound` for now (if the string passed to `PackitException` isn't a resource key, it just uses it as the error message).

Reply:
Added a ticket for this: https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-6221/Interpolated-error-messages